core/src/main/kotlin/com/justai/jaicf/channel/invocationapi/InvocableBotChannel.kt

```diff
@@ -0,0 +1,81 @@
+package com.justai.jaicf.channel.invocationapi
+
+import com.justai.jaicf.channel.http.HttpBotRequest
+import com.justai.jaicf.channel.http.asHttpBotRequest
+import com.justai.jaicf.context.RequestContext
+import com.justai.jaicf.helpers.logging.WithLogger
+import java.util.*
+import kotlin.random.Random
+
+/**
+ * Base class for all channels able to process requests from external service.
+ * Allows to send [InvocationEventRequest] or [InvocationQueryRequest] with client identifier to implementations at any time.
+ *
+ * @see InvocationRequest
+ *
+ * */
+interface InvocableBotChannel : WithLogger {
+    /**
+     * Processes an [InvocationRequest]
+     *
+     * @param request an [InvocationRequest]
+     * @param requestContext additional general request's data that can be used during the request processing
+     *
+     * @see InvocableBotChannel
+     */
+    fun processExternalInvocation(request: InvocationRequest, requestContext: RequestContext)
+
+    /**
+     * Provides a messageId for substitution in request template
+     * */
+    fun provideMessageId(): String = UUID.randomUUID().toString()
+
+    /**
+     * Provides a timestamp for substitution in request template
+     * */
+    fun provideTimestamp(): String = System.currentTimeMillis().toString()
```

Comment:
Why this and function above should be a part of the interface? It's template-specific data that should be calculated during every template rendering process, isn't it?